sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -92,31 +95,57 @@ object StringUtils {
 
   /**
    * Concatenation of sequence of strings to final string with cheap append method
-   * and one memory allocation for the final string.
+   * and one memory allocation for the final string.  Can also bound the final size of
+   * the string.
    */
-  class StringConcat {
+  class StringConcat(val maxLength: Int = Integer.MAX_VALUE) {
```

Comment:
You generally cannot allocate an array of `Integer.MAX_VALUE` elements. Can you use `ByteArrayMethods.MAX_ROUNDED_ARRAY_LENGTH` instead.